website/docs/essays/faq.md

```diff
@@ -75,14 +75,49 @@ floor(R / L)
 
 ## How do you calculate transaction fee?
 
-The size of a normal two-in-two-out transaction is 597 bytes, to calculate transaction fee we need add extra 4 bytes size due to the cost of serialized tx in a block.
+Let's start with a brief introduction to Transaction Weight.
+
+### Transaction Weight
+
+The miners select transactions to fill the limited block space which gives the highest fee. Because there are two different limits, serialized size and consumed cycles, the selection algorithm is a multi-dimensional knapsack problem. Introducing the Transaction weight converts the multi-dimensional knapsack to a typical knapsack problem.
+
+```
+/// Equal to MAX_BLOCK_BYTES / MAX_BLOCK_CYCLES, see [glossary](https://docs.nervos.org/docs/basics/glossary).
+pub const BYTES_PER_CYCLES: f64 = 0.000_170_571_4_f64;
+
+get_transaction_weight(tx_size: usize, cycles: u64) -> u64 {
+    max(
+        tx_size as u64,
+        (cycles as f64 * BYTES_PER_CYCLES) as u64,
+    )
+}
+```
+
+### Estimate cycles
+
+The cycles of the transaction can be obtained via rpc [estimate_cycles](https://github.com/nervosnetwork/ckb/tree/master/rpc#method-estimate_cycles)
+
+Here depends on the type of transaction to be built, if the transaction consists of small cycles of scripts, you can disregard cycles and directly replace weight with tx_size to calculate the transaction fee
 
-`(tx_size + 4) * fee_rate / 1000`
+```
+(tx_size + 4) * fee_rate / 1000
+```
+
+if the transaction consists of large cycles of script, then you need to include cycles in the calculation of the fee, otherwise the fee rate of the transaction will not meet the expectations, of course, it also depends on the scenario, if the priority of the transaction confirm does not need so precise control, you can also directly use a rough estimate of the cycles, or do not consider cycles can also be, according to your own needs trade-offs.
```

Comment:
```suggestion
If the transaction consists of large cycles of script, then you need to include cycles in the calculation of the fee, otherwise the fee rate of the transaction will not meet the expectations, of course, it also depends on the scenario, if the priority of the transaction confirm does not need so precise control, you can also directly use a rough estimate of the cycles, or do not consider cycles can also be, according to your own needs trade-offs.
```